### PR TITLE
[MIG] purchase_batch_invoicing: Migration to 11.0

### PR DESCRIPTION
**Pull request by chienandalu**
This module extends the functionality of purchases to support batch invoicing
purchase orders and to allow you to choose if you want them grouped by purchase
order or by vendor.

cc @Tecnativa

### CODE REVIEW

File: purchase_batch_invoicing/data/ir_cron_data.xml [hunk 1]
```diff
@@ -4,14 +4,14 @@
...
-    <field name="active" eval="False"/>
-    <field name="interval_type">weeks</field>
-    <field name="numbercall">-1</field>
-    <field name="model">purchase.batch_invoicing</field>
-    <field name="function">cron_invoice_all_pending</field>
-    <field name="args">("partner_id",)</field>
-</record>
+    <record id="cron" model="ir.cron">
+        <field name="name">Invoice all pending purchase orders</field>
+        <field name="active">True</field>
```
Comment (pedrobaeza) [inline purchase_batch_invoicing/data/ir_cron_data.xml]: Please don't change this field. Leave it disabled (and it has to be evaluated with `eval`).

File: purchase_batch_invoicing/readme/ROADMAP.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+It would be nice to be able to group invoices by PO line.
```
Comment (pedrobaeza) [inline purchase_batch_invoicing/readme/ROADMAP.rst]: Put `* ` to start bullet list (although with only one element).

File: purchase_batch_invoicing/readme/USAGE.rst [hunk 3]
```diff
@@ -0,0 +1,20 @@
+To use this module, you need to:
+
+#. Have *Purchase / User* permissions.
```
Comment (pedrobaeza) [inline purchase_batch_invoicing/readme/USAGE.rst]: This point should go to `Configuration` section IMO.

File: purchase_batch_invoicing/tests/test_invoicing.py [hunk 4]
```diff
@@ -75,10 +75,13 @@ def setUpClass(cls):
         for po in cls.pos:
             # Confirm purchase order
             po.button_confirm()
-            # Receive products
-            cls.env["stock.immediate.transfer"] \
-                .with_context(active_id=po.picking_ids.id) \
-                .create(dict()).process()
+        # Receive products
+        for picking in cls.pos.mapped('picking_ids'):
+            picking.force_assign()
```
Comment (pedrobaeza) [inline purchase_batch_invoicing/tests/test_invoicing.py]: This shouldn't be needed on receptions

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: purchase_batch_invoicing/tests/test_invoicing.py [hunk 5]
```diff
@@ -1,15 +1,15 @@
-# -*- coding: utf-8 -*-
 # Copyright 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 import mock
```
Comment (yajo) [inline purchase_batch_invoicing/tests/test_invoicing.py]: AFAIK in py3 it is `from unittest import mock`

**Review by yajo — APPROVED** [review]

**Comment by chienandalu** [comment]
@pedrobaeza Changes done

**Review by pedrobaeza — APPROVED** [review]

**Comment by rafaelbn** [comment]
@chienandalu take care of https://github.com/OCA/account-invoicing/pull/384 in v9 that affects in v11 too

**Comment by pedrobaeza** [comment]
Already done in #505, so don't duplicate efforts.

**Pull request merged**